lib/iris/tests/unit/coords/test_CellMeasure.py

```diff
@@ -72,24 +72,6 @@ def test_set_data__lazy(self):
         self.measure.data = new_vals
         self.assertArrayEqual(self.measure.data, new_vals)
 
-    def test_set_data__int__lazy(self):
-        new_vals = as_lazy_data(np.array((1, 2, 3, 4), dtype=np.int32))
-        exp_emsg = "Cannot create cell measure with lazy data of type int32"
-        with self.assertRaisesRegexp(ValueError, exp_emsg):
-            self.measure.data = new_vals
-
-    def test_set_data__uint__lazy(self):
-        new_vals = as_lazy_data(np.array((1, 2, 3, 4), dtype=np.uint32))
-        exp_emsg = "Cannot create cell measure with lazy data of type uint32"
-        with self.assertRaisesRegexp(ValueError, exp_emsg):
-            self.measure.data = new_vals
-
-    def test_data_different_shape(self):
-        new_vals = np.array((1., 2., 3.))
-        msg = 'New data shape must match existing data shape.'
-        with self.assertRaisesRegexp(ValueError, msg):
-            self.measure.data = new_vals
-
```

Comment:
I removed this when I replaced the CellMeasure specific `_data_setter` with the new `_DimensionalMetadata._values_setter` which does not force the new shape to match the old shape. It seemed a bit of a strange thing to enforce, especially considering we don't enforce this requirement on Auxiliary coordinates. 
You could, however, argue that this is a breaking change. In which case I will reinstate this test and add a `data` property to `CellMeasure` that does the shape check as well as calling `_DimensionalMetadata._values_setter`